### PR TITLE
[Alluxio 2255] Improve style in AbstractClientTest

### PR DESCRIPTION
**Pull request by schernolyas**
https://alluxio.atlassian.net/browse/ALLUXIO-2255

### CODE REVIEW

**Comment by calvinjia** [comment]
Jenkins, test this please.

File: core/server/src/test/java/alluxio/AbstractClientTest.java [hunk 1]
```diff
@@ -57,14 +64,12 @@ public void checkVersion(AlluxioService.Client thriftClient, long version) throw
...
-      client.checkVersion(thriftClient, 0);
-      Assert.fail("checkVersion() should fail");
-    } catch (IOException e) {
-      Assert.assertEquals(ExceptionMessage.INCOMPATIBLE_VERSION.getMessage(SERVICE_NAME, 0, 1),
-          e.getMessage());
-    }
+    mExpectedException.expect(IOException.class);
+    mExpectedException.expectMessage(ExceptionMessage.INCOMPATIBLE_VERSION.getMessage(SERVICE_NAME,
+        0, 1));
+    TestClient client = new TestClient();
```
Comment (calvinjia) [inline core/server/src/test/java/alluxio/AbstractClientTest.java]: Could you close the `TestClient()` in a finally?

File: core/server/src/test/java/alluxio/AbstractClientTest.java [hunk 2]
```diff
@@ -57,14 +64,12 @@ public void checkVersion(AlluxioService.Client thriftClient, long version) throw
...
-    try (TestClient client = new TestClient()) {
-      client.checkVersion(thriftClient, 0);
-      Assert.fail("checkVersion() should fail");
-    } catch (IOException e) {
-      Assert.assertEquals(ExceptionMessage.INCOMPATIBLE_VERSION.getMessage(SERVICE_NAME, 0, 1),
-          e.getMessage());
-    }
+    mExpectedException.expect(IOException.class);
+    mExpectedException.expectMessage(ExceptionMessage.INCOMPATIBLE_VERSION.getMessage(SERVICE_NAME,
+        0, 1));
```
Comment (calvinjia) [inline core/server/src/test/java/alluxio/AbstractClientTest.java]: Could you update the formatting here? I think we would want to have a new line after `expectMessage(`

File: core/server/src/test/java/alluxio/AbstractClientTest.java (hunk 1 above)
Comment (schernolyas) [inline core/server/src/test/java/alluxio/AbstractClientTest.java]: Hi!
Ok. I will correct it.

Comment (schernolyas) [inline core/server/src/test/java/alluxio/AbstractClientTest.java]: I produced new commit.
Check it, please

File: core/server/src/test/java/alluxio/AbstractClientTest.java [hunk 3]
```diff
@@ -57,13 +66,12 @@ public void checkVersion(AlluxioService.Client thriftClient, long version) throw
   public void unsupportedVersion() throws Exception {
     final AlluxioService.Client thriftClient = Mockito.mock(AlluxioService.Client.class);
     Mockito.when(thriftClient.getServiceVersion()).thenReturn(1L);
+    mExpectedException.expect(IOException.class);
+    mExpectedException.expectMessage(INCOMPATIBLE_VERSION.getMessage(SERVICE_NAME, 0, 1));
 
-    try (TestClient client = new TestClient()) {
+    try (TestClient client = new TestClient();) {
```
Comment (calvinjia) [inline core/server/src/test/java/alluxio/AbstractClientTest.java]: You shouldn't need the `;` here.

**Comment by calvinjia** [comment]
@schernolyas Looks good overall, I have 1 minor style comment.

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**